### PR TITLE
Add Support for Custom Template Definition

### PR DESCRIPTION
**Pull request by pcalley**
Allows user to supply repo specific overrides of the markdown comment templates. 

1. Templates are checked into atlantis-private and then referenced by their path in the repo_config.yaml. e.g.
```
repos:
  - id: /github.com/*/
    apply_requirements: [mergeable, undiverged]
    template_overrides: 
      project_plan_success: server/events/templates/planSuccessWrappedCustomTmpl.tmpl
```
2. The templates are specified using the same name as the default template in: [markdown_renderer.go](https://github.com/lyft/atlantis/blob/release-v0.17.3-lyft.1/server/events/markdown_renderer.go)

### CODE REVIEW

**Comment by pcalley** [comment]
ptal #orchestration

**Review by nishkrishnan — COMMENTED** [review]
I think this a good start.

I know the description of the ticket was vague but I was sort of hoping for a couple things:

1. something a bit simpler for the end user. There's a million different templates that have a ton of duplication in them so would be great if there was just like one single template for each operation type (plan, apply, policy check) which would be a bit more straightforward.
2. a cleaner markdown_renderer.go file which is currently a mess, I think we sort of added to the mess in this PR, but with this task I wanted to build these template strings in a smarter manner without duplicated template strings, or a million case statements, etc.

File: server/events/command_context.go [hunk 1]
```diff
@@ -49,4 +49,6 @@ type CommandContext struct {
 	PullStatus *models.PullStatus
 
 	Trigger CommandTrigger
+
+	TemplateOverrides map[string]string
```
Comment (nishkrishnan) [inline server/events/command_context.go:53]: This doesn't need to be in the command context as it's not changing between requests, this should probably just be injected into the MarkdownRenderer as part of its construction on server startup.

**Review by nishkrishnan — COMMENTED** [review]
Almsot there.

File: server/events/markdown_renderer_test.go [hunk 2]
```diff
@@ -60,7 +60,7 @@ func TestRenderErr(t *testing.T) {
 		}
 		for _, verbose := range []bool{true, false} {
 			t.Run(fmt.Sprintf("%s_%t", c.Description, verbose), func(t *testing.T) {
-				s := r.Render(res, c.Command, "log", verbose, models.Github)
+				s := r.Render(res, c.Command, "log", verbose, models.Github, make(map[string]string))
```
[the remark this answers is not in the document]
Comment (pcalley) [reply server/events/markdown_renderer_test.go:214]: Added

File: server/events/command_context.go (hunk 1 above)
Comment (pcalley) [reply server/events/command_context.go]: Agreed. Removed this from the command context.

**Comment by pcalley** [comment]
Added a few docs as well.

**Comment by nishkrishnan** [comment]
you'll have to fix these merge conflicts.

**Review by nishkrishnan — APPROVED** [review]

**Pull request merged**